pkg/controller/operators/catalog/subscription/reconciler.go

```diff
@@ -91,11 +91,13 @@ func (c *catalogHealthReconciler) Reconcile(ctx context.Context, in kubestate.St
 
 				var healthUpdated, deprecationUpdated bool
 				next, healthUpdated = s.UpdateHealth(c.now(), catalogHealth...)
-				deprecationUpdated, err = c.updateDeprecatedStatus(ctx, s.Subscription())
-				if err != nil {
-					return next, err
+				if healthUpdated {
+					if _, err := c.client.OperatorsV1alpha1().Subscriptions(ns).UpdateStatus(ctx, s.Subscription(), metav1.UpdateOptions{}); err != nil {
+						return nil, err
+					}
```

Comment:
I was curious about the reasoning behind separating the `UpdateStatus` calls, then realized that making sure the health status update happens error-free before running the deprecation check+update is more important than potentially saving us an extra `UpdateStatus` call. Is that right?

Reply:
@dtfranz so that is why I made this change while I was reading this code, but I've removed the change from this PR now. It go in by mistake 😅; it's an irrelevant change that should be a PR of it's own. Which will most likely happen soon since there's a bug that's been reported in the `update deprecated status` area, so I can make this change when I write up the PR for that bug.

Reply:
Thanks for explaining! That makes sense, since this change is pretty tangential to this PR's objectives.

Reply:
@dtfranz actually, read through it again, and looks like it didn't get in by mistake 😅. The `Deprecated status` area runs into a "could not find registry server for CatalogSource problem"  in [these tests](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/test/e2e/subscription_e2e_test.go#L730-L1029) with the introduction of the changes in this PR. Possibly because of the use of the cache.SourceProvider in its implementation. 

I remember adding this change because it's the path of least resistance: we get the error surfaced to the Catalog status, as attested by the tests "Surface catalog health when there is a problem", and 

`
making sure the health status update happens error-free before running the deprecation check+update is more important than potentially saving us an extra UpdateStatus call
`

Reply:
I added it back in PTAL